frontend/app/components/api-users-table-row.js

```diff
@@ -0,0 +1,66 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import ENV from "../config/environment";
+
+export default class ApiUsersTableRow extends Component {
+  @service store;
+
+  validityTimes = [
+    { label: "profile.api_users.options.one_min", value: 60 },
+    { label: "profile.api_users.options.five_mins", value: 300 },
+    { label: "profile.api_users.options.twelve_hours", value: 43200 },
+    { label: "profile.api_users.options.infinite", value: 0 }
+  ];
+
+  get selectedValidFor() {
+    return this.validityTimes.find(
+      (t) => t.value == this.args.apiUser.validFor
+    );
+  }
+
+  @action
+  toggleApiUser(user) {
+    let httpMethod = user.locked ? "delete" : "post";
+    /* eslint-disable no-undef  */
+    return fetch(`/api/api_users/${user.id}/lock`, {
+      method: httpMethod,
+      headers: {
+        "Content-type": "application/x-www-form-urlencoded; charset=UTF-8",
```

Comment:
we're using this fetch method with the same headers over and over again.
would it make sense to define a global custom fetch method? #make-it-dry